### PR TITLE
Update operator to v0.39.0 and enable disableMountSubpath

### PR DESCRIPTION
**Pull request by aiyengar2**
Changes required:
- Updating all the CRDs & RBAC to match https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.39/bundle.yaml
- Removing deprecated fields of `manageCRDs` and ` withValidation`
- Set `disableMountSubpath` for Alertmanager and Prometheus to true by default.

#### Why didn't you upgrade to the latest version of Prometheus Operator?

Although the current latest is `v0.42.1`, the upstream [`kube-prometheus-stack`](https://github.com/prometheus-community/helm-charts/blob/5b25a9b68f9327535a3b044139549a3bfcfc88a5/charts/kube-prometheus-stack/values.yaml#L1349) Helm chart hasn't even upgraded past `v0.38.1` yet and, as a result, this version of Prometheus Operator isn't even supported by Monitoring V2 yet. To avoid introducing changes that could introduce problems that haven't even been addressed by the upstream chart yet, I chose to make the minimal upgrade to `v0.39.0` to just expose the requested `disableMountSubpath` fields.

Related Issue: https://github.com/rancher/rancher/issues/29149
Depends on:
- [ ] Add Prometheus Operator v0.39.0 images (https://github.com/rancher/image-mirror/pull/40)

### CODE REVIEW

File: charts/rancher-monitoring/v0.1.3/charts/operator/templates/deployment.yaml
Comment (aiyengar2) [inline charts/rancher-monitoring/v0.1.3/charts/operator/templates/deployment.yaml:52]: Part of v0.39.0 changes removed these fields: https://github.com/prometheus-operator/prometheus-operator/blame/99ecd7100c2dd9c1392e7670215045cf41606a0f/CHANGELOG.md#L68

File: charts/rancher-monitoring/v0.1.3/charts/operator-init/files/crd-prometheusrules.yaml
Comment (aiyengar2) [inline charts/rancher-monitoring/v0.1.3/charts/operator-init/files/crd-prometheusrules.yaml:2]: They switched from `apiextensions.k8s.io/v1beta` to `apiextensions.k8s.io/v1` as part of https://github.com/prometheus-operator/prometheus-operator/blame/99ecd7100c2dd9c1392e7670215045cf41606a0f/CHANGELOG.md#L70.

This also requires users who deploy this chart to be at minimum in k8s 1.16

**Review by aiyengar2 — COMMENTED** [review]
Based on the [CHANGELOG.md](https://github.com/prometheus-operator/prometheus-operator/blame/99ecd7100c2dd9c1392e7670215045cf41606a0f/CHANGELOG.md#L70), this version of monitoring can only be supported in k8s v1.16+. Is there any way to specify this in the chart (i.e. in the `questions.yml`)?

File: charts/rancher-monitoring/v0.1.3/values.yaml
Comment (aiyengar2) [inline charts/rancher-monitoring/v0.1.3/values.yaml:10]: https://github.com/rancher/image-mirror/pull/40

**Review by aiyengar2 — COMMENTED** [review]
Blocked on merging this till images are created:
- [ ] https://github.com/rancher/image-mirror/pull/40

File: charts/rancher-monitoring/v0.1.3/values.yaml
Comment (aiyengar2) [inline charts/rancher-monitoring/v0.1.3/values.yaml]: I need this to test locally till the images are created

File: charts/rancher-monitoring/v0.1.3/charts/operator/templates/deployment.yaml
Comment (pennyscissors) [reply charts/rancher-monitoring/v0.1.3/charts/operator/templates/deployment.yaml:52]: Yes, you can use `rancher_min_version: <2.x.x version>` where 2.x.x version is the minimum version of rancher that supports v1.16+. You can use the following to figure that out:
https://rancher.com/support-maintenance-terms/all-supported-versions/rancher-v2.3.6/

Example questions.yaml:
https://github.com/rancher/charts/blob/6aaa35cacac454348ef9dce75f8e8c2508137663/charts/longhorn/v0.4.1/questions.yml#L1

**Review by aiyengar2 — COMMENTED** [review]
> Blocked on merging this till images are created:
> 
> * [ ]  [rancher/image-mirror#40](https://github.com/rancher/image-mirror/pull/40)

Last commit will be dropped before merge

File: charts/rancher-monitoring/v0.1.3/charts/operator/templates/deployment.yaml
Comment (pennyscissors) [reply charts/rancher-monitoring/v0.1.3/charts/operator/templates/deployment.yaml:52]: oops replied to the wrong comment, but you got it anyways 😁

**Review by pennyscissors — APPROVED** [review]
lgtm

File: charts/rancher-monitoring/v0.1.3/charts/operator/templates/deployment.yaml
Comment (aiyengar2) [reply charts/rancher-monitoring/v0.1.3/charts/operator/templates/deployment.yaml:52]: Hmm so currently this minimum version is set to: https://github.com/rancher/system-charts/blob/dc3df5b5342b8210861bf800420589117feafc89/charts/rancher-monitoring/v0.1.3/questions.yml#L1.

But this change needs to be backported to be supported in either/both `2.3.8+` or `2.4.6+`, both of which currently support k8s 1.15 based on the doc that you sent.

**Comment by aiyengar2** [comment]
#### Potential Block on Merging

As discussed in https://github.com/rancher/system-charts/pull/345#discussion_r494636586, these changes will fail on a cluster that doesn't have k8s 1.16 installed and there's currently no Rancher release that exclusively supports k8s 1.16+.

I tried these changes out on a k8s 1.15 cluster and saw the following which confirms that these changes would fail to deploy on a k8s 1.15 cluster:
screen_shot_2020-09-24_at_4 53 14_pm

Since the `operator-init-crds` deployment fails, both Monitoring V1 and Alerting V1 would be unable to be deployed on a k8s 1.15 cluster that tries to use this new version.

Unclear what the path forward is with this PR. I could try to look into making changes to the CRDs themselves (e.g. to replace `apiextensions.k8s.io/v1` with the old `apiextensions.k8s.io/v1beta`), but in previous versions of monitoring we've only ever directly copied the CRDs from what was generated in Prometheus Operator. I'm unsure what the repercussions of changing the CRDs would be on the Prometheus Operator deployment.

@deniseschannon @maggieliu will there be any upcoming releases in Rancher 2.3 / 2.4 that will ever drop support for k8s 1.15? If not, this change might be something we can only support in Monitoring V2 since Monitoring V1 is no longer supported in 2.5+.

**Comment by maggieliu** [comment]
related to: https://github.com/rancher/rancher/issues/29149

**Comment by aiyengar2** [comment]
https://github.com/rancher/rancher/issues/29149#issuecomment-699138151

**Pull request closed without merging**
